### ISSUE TITLE
Redis 6 support

### ISSUE THREAD

**Issue by omnibs** [informatikr/hedis#168]
Hi! Thank you for writing Hedis =]

I noticed the readme says:

> Hedis is intended to be used with the latest stable version of Redis (currently 5.0).

Is the readme merely outdated, or is 6.0 indeed not supported yet?

**Comment by michaelglass** [comment]
from https://raw.githubusercontent.com/antirez/redis/6.0/00-RELEASENOTES

```
Migrating from 5.0 to 6.0
=========================

Redis 6.0 is mostly a strict superset of 5.0, you should not have any problem
upgrading your application from 5.0 to 6.0. However this is a list of small
non-backward compatible changes introduced in the 6.0 release:

* The SPOP <count> command no longer returns null when the set key does not
  exist. Now it returns the empty set as it should and as happens when it is
  called with a 0 argument. This is technically a fix, however it changes the
  old behavior.
```

I'm guessing we can test out redis 6 compatibility and report back?

**Comment by michaelglass** [comment]
looks like tests on master here are already broken so it'll be a little tricky to test.

**Comment by k-bx** [comment]
PRs fixing all of this are very much welcome!

Tests worked for me locally last time I've checked, so it was only Travis that needed some love. Anyways, open for contributions, but don't have a lot of time for this ATM myself, unfortunately.

**Comment by michaelglass** [comment]
preliminary info:
we upgraded our redis _cluster_ from 5.0.6 to 6.0.4
- all existing applications were happy / stayed online / continued to talk to redis
- new apps failed during boot (??!!). will probably debug further

So maybe there's support for a single redis instance but not cluster yet? I couldn't see significant cluster changes in the changelog so a little confusing

**Comment by k-bx** [comment]
Cluster support was added in https://github.com/informatikr/hedis/pull/157 , please see if that's what you're looking for

**Comment by michaelglass** [comment]
Yup! Our team (mostly jasper) wrapped up that cluster work, but we only tested against Redis 5.0 in cluster mode. Upgrading to Redis 6 in cluster mode caused some strange errors we'll try to debug.

**Comment by k-bx** [comment]
Ah, gotcha! Thanks

**Comment by aravindgopall** [comment]
@michaelglass is this still the case. we have successfully tested with `juspay` fork for Redis 6.

**Comment by michaelglass** [comment]
@aravindgopall sorry I'm not on the project anymore. Happy to hear it's working for you.

**Comment by omnibs** [comment]
@aravindgopall thank you so much for reporting, i'll give it a try!

**Comment by qnikst** [comment]
Please see MR #211 it provides redis 7 support for all existing functions. It does not provide support for newly added functions. Unfortunately I see no way to make the change backward compatible, my company adopted those changes, but I would love to see if anyone would give that a try before merging into the master.

**Comment by qnikst** [comment]
@aravindgopall, sorry for asking you here, but I wonder if Juspay want to contribute efforts upstream, otherwise be may get into a situations here the codebases would significally diverge making it impossible to combine features that appear in the projects.

**Comment by asheshambasta** [comment]
We've been using hedis with redis `6.2.6` successfully also. I'd love to know what kind of contributions can be made and open PRs. 

I'd also like to add support for [GETEX](https://redis.io/docs/latest/commands/getex/).

**Comment by qnikst** [comment]
Implementation of all commands from 6.2 is available at https://github.com/informatikr/hedis/pull/254

Planned 0.16.2 release
